annotation/src/main/java/org/cbioportal/genome_nexus/annotation/web/AnnotationController.java

```diff
@@ -382,6 +382,7 @@ private VariantAnnotation getVariantAnnotation(String variant)
                     // this is thrown when the annotationJSON can't be stored by mongo
                     // due to the variant annotation key being too large to index
                     LOG.info(e.getLocalizedMessage());
+                    return variantAnnotation;
```

Comment:
I don't see how this would change behavior. If the flow of control reached this point and there was no return statement here, it seems that we have consumed the DataIntegrityViolationException and would next exit the  outer try block (line 362 - 387) without further exception, so we would skip the two catch blocks below and reach the (identical) return on line 400. Maybe somehow the change to variantAnnotation within the try blocks (line 373) is being undone because of unwinding the inner try block during the DataIntegrityViolation Exception?